src/core/cardano/walletConnect/peerConnection.ts

```diff
@@ -102,12 +114,44 @@ class PeerConnection {
     );
   }
 
-  connectWithDApp(dAppIdentifier: string) {
+  async connectWithDApp(dAppIdentifier: string) {
     if (this.identityWalletConnect === undefined) {
       throw new Error(PeerConnection.PEER_CONNECTION_START_PENDING);
     }
-
+    const existedPeerConnection =
+      await Agent.agent.peerConnectionMetadataStorage
+        .getPeerConnectionMetadata(dAppIdentifier)
+        .catch((error) => {
+          if (
+            error.message ===
+            PeerConnectionStorage.PEER_CONNECTION_METADATA_RECORD_MISSING
+          ) {
+            return undefined;
+          } else {
+            throw error;
+          }
+        });
+    if (!existedPeerConnection) {
+      await Agent.agent.peerConnectionMetadataStorage.createPeerConnectionMetadataRecord(
+        {
+          id: dAppIdentifier,
+          iconB64: ICON_BASE64,
+        }
+      );
+    } else {
+      //Set other fields back to pending state
+      await Agent.agent.peerConnectionMetadataStorage.updatePeerConnectionMetadata(
```

Comment:
I don't think we need this else block - we can just let it stay as whatever it was before until we connect